### PR TITLE
[RF] Code modernization of RooAbsCachedPdf

### PR DESCRIPTION
**Pull request by guitargeek**
The usual RooFit code modernizations.

### CODE REVIEW

File: roofit/roofitcore/src/RooAbsCachedPdf.cxx [hunk 1]
```diff
@@ -292,29 +244,24 @@ RooAbsCachedPdf::PdfCacheElem::PdfCacheElem(const RooAbsCachedPdf& self, const R
...
+      name += arg->GetName();
     }
-    delete iter ;
   }
 
-  name.Append("]") ;
-  const char* payloadUS = payloadUniqueSuffix() ;
-  if (payloadUS) {
-    name.Append(payloadUS) ;
+  name += "]";
```
Comment (eguiraud) [inline roofit/roofitcore/src/RooAbsCachedPdf.cxx:262]: maybe

```suggestion
  name += ']';
```

File: roofit/roofitcore/src/RooAbsCachedPdf.cxx [hunk 2]
```diff
@@ -292,29 +244,24 @@ RooAbsCachedPdf::PdfCacheElem::PdfCacheElem(const RooAbsCachedPdf& self, const R
...
+  std::string name = "_Obs[";
+  if (!nset.empty()) {
+    bool first(true) ;
+    for(auto const& arg : nset) {
       if (first) {
-	first=kFALSE ;
+        first=false ;
       } else {
-	name.Append(",") ;
+        name += ",";
```
Comment (eguiraud) [inline roofit/roofitcore/src/RooAbsCachedPdf.cxx:256]: maybe consider
```suggestion
        name += ',';
```

**Review by eguiraud — APPROVED** [review]
LGTM, nothing to say! :D

File: roofit/roofitcore/src/RooAbsCachedPdf.cxx (hunk 2 above)
Comment (guitargeek) [reply roofit/roofitcore/src/RooAbsCachedPdf.cxx:256]: Okay! I'll update.

**Pull request merged**